### DISCUSSION TITLE
Flux v2 migration on GKE cluster with Workload Identity enabled

### DISCUSSION THREAD

**Discussion by trenslow** [fluxcd/flux2#1492] [Q&A] [answered]
Hi community,

I'm attempting to migrate to Flux v2 and I'm stuck on the image automation part. Maybe you can point me in the right direction.

I'm deploying Flux v2 in a Google GKE cluster with Workload Identity. On Flux v1, everything worked if I followed the docs. However, I'm a bit lost on how to configure Flux v2 to use the existing Google Service account.

My thought was to kustomize the ServiceAccount for the image-reflector-controller with the annotation necessary to enable authentication to GCR via Workload Identity, as it's the one complaining about not being allowed access. The error looks like the following:
`{"level":"error","ts":"2021-06-09T09:07:38.990Z","logger":"controller-runtime.manager.controller.imagerepository","msg":"Reconciler error","reconciler group":"image.toolkit.fluxcd.io","reconciler kind":"ImageRepository","name":"<REDACTED>","namespace":"flux-system","error":"GET https://eu.gcr.io/v2/<REDACTED>/<REDACTED>/tags/list?n=1000: DENIED: Failed to read tags for host 'eu.gcr.io', repository '/v2/<REDACTED>/<REDACTED>/tags/list?n=1000'"}`

The patch went through successfully as I can see the annotation on the reflector controller's ServiceAccount, but the controller still gets denied access to GCR. I double-checked that my IAM policies are properly setup for workload identity and that the service account has permission to list tags.

What am I missing here?

**Comment by stefanprodan** [comment, marked as answer]
Check out the docs: https://fluxcd.io/docs/guides/image-update/#gcp-container-registry

**Reply by trenslow** [reply]
Sorry about that, I must have overlooked that doc. Thanks for the help!

**Reply by trenslow** [reply]
Just out of curiosity: how come it's not possible to add the annotation to the image-reflector controller ServiceAccount to allow it to authenticate to GCR? Some technical limitation/design choice?

**Reply by stefanprodan** [reply]
See https://github.com/fluxcd/image-reflector-controller/issues/139#issuecomment-852000782
